sources/packages/backend/apps/workers/src/services/supporting-user/supporting-user.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from "@nestjs/common";
+import { DataSource } from "typeorm";
+import {
+  RecordDataModelService,
+  Application,
+  SupportingUser,
+  SupportingUserType,
+} from "@sims/sims-db";
+
+@Injectable()
+export class SupportingUserService extends RecordDataModelService<SupportingUser> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(SupportingUser));
+  }
+
+  /**
+   * Checks if the supporting users are already present for an application.
+   * @param applicationId application to be verified.
+   * @returns true if the application already have the supporting users,
+   * otherwise, false.
+   */
+  async hasSupportingUsers(applicationId: number) {
+    const someSupportingUser = await this.repo.findOne({
+      select: { id: true },
+      where: { application: { id: applicationId } },
+    });
+    return !!someSupportingUser;
+  }
+
+  /**
+   * Get the supporting user data by its id.
+   * @param supportingUserId supporting user id.
+   * @returns supporting user data.
+   */
+  async getSupportingUserById(
+    supportingUserId: number,
+  ): Promise<SupportingUser> {
+    return this.repo.findOne({
+      select: { id: true, supportingData: true },
+      where: { id: supportingUserId },
+    });
+  }
+
+  /**
+   * Create the supporting users of provided types.
+   * @param applicationId application to have the supporting users created.
+   * @param supportingUserTypes types to be create. This method accepts, for instance,
+   * - ['Parent', 'Parent']: two parents will be created;
+   * - ['Partner', 'Parent']: one parent and one partner will be created.
```

Comment:
As an example [Parent,Parent,Partner] is also possible right?

Reply:
From the current business perspective and current workflow implementation, it will be `[Parent, Parent]` or `[Partner]`.
The idea of adding the examples was to show what the method can do. The method is supposed to be generic and capable of create supporting users as needed.
I changed the comment as below. @ann-aot also had some questions about it. Please let me know if it is clear.
```ts
/**
   * Create the supporting users of provided types.
   * @param applicationId application to have the supporting users created.
   * @param supportingUserTypes types to be created. This method accepts, for instance,
   * - ['Parent', 'Parent']: two parents will be created;
   * - ['Partner']: one partner will be created.
   * - ['Partner', 'Parent']: one parent and one partner will be created.
   * - ['Parent', 'Parent', 'Partner', 'Partner']: multiple parents and multiples partners.
   * These examples represent what this method can do and are not necessarily
   * the current business logic.
   * @returns created supporting users.
   */
```